server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -202,6 +214,16 @@ public void addMaxIndexSizeCondition(ByteSizeValue size) {
         this.conditions.put(maxSizeCondition.name, maxSizeCondition);
     }
 
+    /**
+     * Adds a size-based condition to check if the size of the largest primary shard is at least <code>size</code>.
+     */
+    public void addMaxIndexSinglePrimarySizeCondition(ByteSizeValue size) {
```

Comment:
Same here about how I think we can drop `Index` and just go with `addMaxSinglePrimarySizeCondition`